org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/managers/AbstractProjectsManagerBasedTest.java

```diff
@@ -268,24 +269,27 @@ public void cleanUp() throws Exception {
 		projectsManager = null;
 		Platform.removeLogListener(logListener);
 		logListener = null;
+		Job.getJobManager().setProgressProvider(null);
 		try {
 			waitForBackgroundJobs();
 		} catch (Exception e) {
 			JavaLanguageServerPlugin.logException(e);
 		}
 		WorkspaceHelper.deleteAllProjects();
+		try {
+			waitForBackgroundJobs();
+		} catch (Exception e) {
+			JavaLanguageServerPlugin.logException(e);
+		}
+		File workspaceDir = new File("target", "workingProjects");
 		try {
 			// https://github.com/eclipse/eclipse.jdt.ls/issues/996
-			FileUtils.forceDelete(getWorkingProjectDirectory());
+			FileUtils.forceDelete(workspaceDir);
```

Comment:
Why is this change needed ? `getWorkingProjectDirectory()` should perform the same. Only difference is in forcibly creating the directory, but that shouldn't happen.

Reply:
> Only difference is in forcibly creating the directory, but that shouldn't happen.

Right!

Reply:
@snjeza Is there a reason not to leave this as it was before ?

Reply:
@rgrunber Yes, it is. It creates workingProjects for no reason.